plugins/user-settings/src/components/FeatureFlags/UserSettingsFeatureFlags.tsx

```diff
@@ -59,15 +68,67 @@ export const UserSettingsFeatureFlags = () => {
     return <EmptyFlags />;
   }
 
+  const clearFilterInput = () => {
+    setFilterInput('');
+    inputRef?.current?.focus();
+  };
+
+  let filteredFeatureFlags = Array.from(featureFlags);
+
+  const filterInputParts = filterInput
+    .split(/\s/)
+    .map(part => part.trim().toLowerCase());
+
+  filterInputParts.forEach(
+    part =>
+      (filteredFeatureFlags = filteredFeatureFlags.filter(featureFlag =>
+        featureFlag.name.toLowerCase().includes(part),
+      )),
+  );
+
+  const Header = () => (
+    <Grid container style={{ justifyContent: 'space-between' }}>
+      <Grid item xs={6} md={8}>
+        <Typography variant="h5">Feature Flags</Typography>
+      </Grid>
+      {featureFlags.length >= 10 && (
+        <Grid item xs={6} md={4}>
+          <TextField
+            label="Filter"
+            style={{ display: 'flex', justifyContent: 'flex-end' }}
+            inputRef={ref => ref && ref.focus()}
+            InputProps={{
+              ...(filterInput.length && {
+                endAdornment: (
+                  <React.Fragment>
+                    <IconButton
+                      aria-label="clear filterInput's value"
+                      onClick={clearFilterInput}
+                      edge="end"
+                    >
+                      <ClearIcon />
+                    </IconButton>
+                  </React.Fragment>
+                ),
+              }),
+            }}
+            onChange={e => setFilterInput(e.target.value)}
+            value={filterInput}
+          />
+        </Grid>
+      )}
+    </Grid>
+  );
+
   return (
-    <InfoCard title="Feature Flags">
+    <InfoCard title={<Header />}>
       <List dense>
-        {featureFlags.map(featureFlag => {
+        {filteredFeatureFlags.map((featureFlag, index) => {
           const enabled = Boolean(state[featureFlag.name]);
 
           return (
             <FlagItem
-              key={featureFlag.name}
+              key={index + 1}
```

Comment:
I think this was actually right before this - the name was probably a perfect key to use, since the key is used to detect items moving around and saving re-renders. Did you change this for some particular reason?